### PR TITLE
Calculate various ASR related indicators in one go

### PR DESCRIPTION
**Pull request by sravfeyn**
@emord 
This should allow ASR report to load this UCR once instead of 7 times. I will update the block ASR config, after we tally the indicators between the UCR report and respective section in ASR report.

### CODE REVIEW

**Review by emord — CHANGES_REQUESTED** [review]
Very pro this, but it needs to be moved to a separate report because this is referenced in mobile reports.

Can you also create an `asr/dashboard` folder for this to go into https://github.com/dimagi/commcare-hq/tree/master/custom/icds_reports/ucr/reports ?

Does this need to also include a change to block_asr.json?

**Comment by sravfeyn** [comment]
Cool, moved it to a separate file.

> Does this need to also include a change to block_asr.json?

Since it updates the data model, the team might want some QA before it gets released. So, I will do this separately.

**Comment by sravfeyn** [comment]
Actually we can release this UCR under a toggle for QA. Updated

File: corehq/toggles.py [hunk 1]
```diff
@@ -1744,3 +1744,10 @@ def _commtrackify(domain_name, toggle_is_enabled):
     TAG_CUSTOM,
     [NAMESPACE_DOMAIN, NAMESPACE_USER]
 )
+
+IMPROVED_ASR_REPORT = StaticToggle(
```
Comment (emord) [inline corehq/toggles.py]: Any reason to not add this to the other toggle you created in  a previous PR?

**Review by emord — APPROVED** [review]

File: corehq/toggles.py (hunk 1 above)
Comment (sravfeyn) [reply corehq/toggles.py]: This will allow independent release, QA for both.

**Comment by sravfeyn** [comment]
We tested this out on ICDS. This actually resulted in a performance drop than improvement. I don't have a solid reason on why this might be, but it's probably because in this way, more data is returned from SQL which is filtered/aggregated in Python, as opposed to the other way where because of the SQL filters smaller amount of rows have to be processed in Python. I think this might be only useful if we somehow cut down the extra duplicate queries that are triggered in for other subreports.

I am going to revert this for now.

**Pull request merged**